external/avro/src/main/scala/org/apache/spark/sql/avro/AvroDataToCatalyst.scala

```diff
@@ -80,22 +80,36 @@ private[avro] case class AvroDataToCatalyst(
   }
 
   @transient private lazy val nullResultRow: Any = dataType match {
-      case st: StructType =>
-        val resultRow = new SpecificInternalRow(st.map(_.dataType))
-        for(i <- 0 until st.length) {
-          resultRow.setNullAt(i)
-        }
-        resultRow
+    case st: StructType =>
+      val resultRow = new SpecificInternalRow(st.map(_.dataType))
+      for (i <- 0 until st.length) {
+        resultRow.setNullAt(i)
+      }
+      resultRow
 
-      case _ =>
-        null
-    }
+    case _ =>
+      null
+  }
 
+  private def getBinaryOffset(binary: Array[Byte]): Int = {
+    val MAGIC_ONE = 0xC3.toByte
+    val MAGIC_TWO = 0x01.toByte
+
+    binary match {
```

Comment:
It might be more readable to just have an if statement here checking the first two bytes

Reply:
I'd also leave comments about what this is doing and why